lib/libc/minimal/include/time.h

```diff
@@ -31,8 +31,15 @@ struct tm {
 	int tm_isdst;
 };
 
+#if !defined(__time_t_defined)
```

Comment:
I believe this needs to be pushed down into `sys/_timespec.h` to work with the existing `posix/time.h`.

It also probably needs to appear in `sys/_timeval.h`.

After that's done it looks like there's some other cleanup that would be appropriate.

@galak would you mind if I took your commit and reworked it for these cases?  It'd be easier than trying to explain.

Reply:
Sure, feel free to create a PR based on top of this one.